ethpm/utils/contract.py

```diff
@@ -0,0 +1,46 @@
+import re
+
+from ethpm.exceptions import ValidationError
+from web3 import Web3
+
+from eth_utils import (
+    to_dict,
+    encode_hex
+)
+
+
+def validate_minimal_contract_data_present(contract_data):
+    """
+    Validate that contract data contains at least one of the following keys
+    necessary to generate contract factory.
+
+    "abi", "bytecode", "runtime_bytecode"
+    """
+    if not any(key in contract_data.keys() for key in ("abi", "bytecode", "runtime_bytecode")):
+        raise ValidationError(
+            "Minimum required contract data (abi/bytecode/runtime_bytecode) not found."
+        )
+
+
+def validate_contract_name(name):
+    pattern = re.compile("^[a-zA-Z][-a-zA-Z0-9_]{0,255}$")
```

Comment:
It's small savings but moving this to a module level definition instead of re-defining it everytime this function runs will yield some albeit minor speed improvements.

Reply:
Went ahead and merged.  I figure we can do this cleanup separately.